src/http/client/client.cr

```diff
@@ -485,6 +511,10 @@ class HTTP::Client
     end
   end
 
+  private def self.allowed_scheme?(uri : URI)
+    uri.scheme =~ /https?/
```

Comment:
can we do something like `starts_with("http")`? I don't know that having an `ssl?` method that checks *all* these conditions is very expressive. For example: `raise "Unsupported scheme: #{uri.scheme}" unless self.ssl?(uri)` could check for http and https as you suggest, but then it reads as though our only valid scheme is "https".

In terms of in-lining, can't we tell the compiler to do this?